src/index.js

```diff
@@ -1000,21 +1000,66 @@ class Enmap extends Map {
   }
 
   /** 
-   * INTERNAL METHOD. Checks if the key is valid.
-   * Will THROW AN ERROR if the key is nil, not a string or not an array of strings.
+   * INTERNAL METHOD. Checks if the key or array of keys is valid.
+   * Will THROW AN ERROR if the key is neither a string nor an array of strings.
    * @param {string|Array<string>} keys The key(s) to check.
    */
   #checkKeyOrArrayOfKeys(keys) {
-    if (!(typeof keys === 'string' || (Array.isArray(keys) && keys.every(e => typeof e === "string")))) {
+    if (Array.isArray(keys)) {
+      this.#checkArrayOfKeys(keys);
+    } else {
+      this.#checkKey(keys);
```

Comment:
Doing this this way means the error won't say it can accept an array of a non-array is passed

Reply:
uh?

Reply:
For example, if I pass a number to a method that may take an array, it will only say it can take a string

Reply:
Yeah for me that's the expected behavior. I think you usually know if a variable is an Array or not. And the error also outputs which type you passed. The message that for example number and string don't align is enough.

Reply:
Ok